app/models/analyticEvent.server.ts

```diff
@@ -1,56 +1,187 @@
-      data: {
-        ip: ip,
-        event: event,
-        data: data,
-      },
-    });
-  }
+  ip: string,
+  event: string,
+  data: string,
+  ts: number
+): Promise<void> {
+  await prisma.analyticEvent.create({
+    data: {
+      ip: ip,
+      event: event,
+      data: data,
+      ts: ts
+    }
+  });
+}
 
 export async function getRecentEvents(): Promise<AnalyticEvent[]> {
   return prisma.analyticEvent.findMany({
     take: 200,
-    orderBy: { ts: "desc" },
+    orderBy: { ts: "desc" }
+  });
+}
+
+export async function getTallyEvents(): Promise<AnalyticEventByDay[]> {
+  return prisma.analyticEventByDay.findMany({
+    take: 200,
+    orderBy: { day: "desc" }
   });
 }
 
+export async function deleteOldEvents(): Promise<void> {
+  const currentDate = new Date();
+  const ninetyDaysAgo = new Date(currentDate);
+  ninetyDaysAgo.setDate(currentDate.getDate() - 90);
+
+  const ninetyDaysAgoUnixTimestamp = Math.floor(ninetyDaysAgo.getTime() / 1000);
+
+  try {
+    await prisma.analyticEvent.deleteMany({
+      where: {
+        ts: {
+          lt: ninetyDaysAgoUnixTimestamp
+        }
+      }
+    });
+  } catch (error) {
+    console.error("Error occurred while deleting 90 day old data: ", error);
+    throw error;
+  }
+}
+
 type AnalyticEventWithCount = AnalyticEvent & { count: number };
 type ARes = {
-  event:string,
-  data:string,
-  count:number
-}
+  event: string;
+  data: string;
+  count: number;
+};
+
+type AResByDay = {
+  event: string;
+  data: string;
+  count: number;
+  ts: string;
+};
 
 export async function getAnalyticsDataLast7Days(): Promise<Array<ARes>> {
   const startDate = new Date();
   startDate.setDate(startDate.getDate() - 7);
+  const startDateUnixTimestamp = Math.floor(startDate.getTime() / 1000);
 
   const analyticsData = await prisma.analyticEvent.groupBy({
     where: {
       ts: {
-        gte: startDate.toISOString(),
-      },
+        gte: startDateUnixTimestamp
+      }
     },
-    by: ['event', 'data', 'ip'],
+    by: ["event", "data", "ip"]
   });
 
-  const map = new Map<string,ARes>();
-  analyticsData.forEach(x => {
-    const key = x.event+x.data;
-    const prev = map.get(key) ?? {event:x.event, data:x.data, count:0};
-    prev.count +=1;
+  const map = new Map<string, ARes>();
+  analyticsData.forEach((x) => {
+    const key = x.event + x.data;
+    const prev = map.get(key) ?? { event: x.event, data: x.data, count: 0 };
+    prev.count += 1;
     map.set(key, prev);
   });
 
   var res = [...map.values()];
   res = res.sort((a, b) => b.count - a.count);
   return res;
-}
\ No newline at end of file
+}
+
+export async function tallyAnalytics(): Promise<Array<AResByDay>> {
+  try {
+    const latestTally = await prisma.analyticEventByDay.findFirst({
+      orderBy: {
+        day: "desc"
+      }
+    });
+
+    let startDate;
+    let endDate;
+
+    if (latestTally) {
+      endDate = new Date(latestTally.day);
+      startDate = new Date();
```

Comment:
if your startDate is the same in either case, you can define it outside the if statement

Reply:
Resolved.